### PR TITLE
Add notification Activation token

### PR DESCRIPTION
**Pull request by 3v1n0**
See https://github.com/flatpak/xdg-desktop-portal/pull/791

### CODE REVIEW

File: src/notification.c [hunk 1]
```diff
@@ -248,12 +280,32 @@ handle_remove_notification (XdpImplNotification *object,
   return TRUE;
 }
 
+static void
+check_notification_impl_version (void)
+{
+  guint activate_signal;
+  GSignalQuery activate_signal_query = {0};
+
+  activate_signal = g_signal_lookup ("action-invoked",
```
Comment (ebassi) [inline src/notification.c:289]: Yuck.

Can't we use a proper version?

File: src/notification.c [hunk 2]
```diff
@@ -146,15 +155,38 @@ activate_action (GDBusConnection *connection,
...
+            }
+
+          signal_name = "ActionInvoked2";
+          invoked_parameters = g_variant_new ("(sss@a{sv}@av)",
+                                              app_id, id, name,
+                                              g_variant_builder_end (&pdata),
+                                              g_variant_builder_end (&parms));
+        }
+      else
+        {
```
Comment (ebassi) [inline src/notification.c:176]: You're missing:

```c
signal_name = "ActionInvoked";
```

File: src/notification.c [hunk 3]
```diff
@@ -146,15 +155,38 @@ activate_action (GDBusConnection *connection,
...
+          invoked_parameters = g_variant_new ("(sss@av)",
+                                              app_id, id, name,
+                                              g_variant_builder_end (&parms));
+        }
+
       g_dbus_connection_emit_signal (connection,
                                      NULL,
                                      "/org/freedesktop/portal/desktop",
                                      "org.freedesktop.impl.portal.Notification",
                                      "ActionInvoked",
```
Comment (ebassi) [inline src/notification.c:186]: You're using the old `ActionInvoked` signal with a different payload.

File: src/notification.c (hunk 1 above)
Comment (3v1n0) [reply src/notification.c:289]: IIRC there was no way to get at this level? Or I wanted to support both :)

I need to check the details...So long time :)

**Comment by TingPing** [comment]
Similar changes landed as https://github.com/flatpak/xdg-desktop-portal-gtk/commit/48bbf07016e75ec949404265816671a9ef62e722

Please open a new PR if anything else was missing.

**Pull request closed without merging**